DirectProgramming/DPC++FPGA/PlatformTests/board_test/src/board_test.hpp

```diff
@@ -0,0 +1,1852 @@
+// bool report_chk - control if Quartus compiled frequency should be read
+// Returns:
+// 0 is test passes, 1 if test fails
+
+// The function does the following tasks:
+// 1. Launches an no-operation NDRange kernel with global size 128 MB
+// 2. Measures time taken for the above NDRange kernel using Sycl event
+// profiling information
+// 3. Obtain kernel clock frequency based on time take for 128 Mglobal
+// operations (NDRange)
+// 4. If the <report_chk> is true, compare the above hardware frequency with
+// Quartus compiled frequency NOTE: This needs the reports or
+// board_test.prj/reports directory from compilation output to be in the same
+// directory as board_test.fpga binary
+// 5. Return 0 (test pass) if measured frequency is within 2 of Quartus compiled
+// frequency, else report error and terminate test NOTE: If <report_chk> is set
+// to false, comparison with Quartus compiled frequency is not done and
+// remaining tests in board_test continue without this frequency check of 2%
+// error tolerance
+
+int ShimMetrics::KernelClkFreq(queue &q, bool report_chk) {
+  std::cout << "*** NOTE ***: This test expects the reports folder generated " 
+            << "during compile to be in one of the following locations : \n"
+            << "    1. Same directory as board_test.fpga binary\n"
+            << "    2. Inside board_test.prj folder in the same directory as "
+            << "board_test.fpga binary\n"
+            << "    If the reports folder is not found, the test will return "
+            << "only the measured kernel clock frequency and fail, none of the "
+            << "other tests will run as hardware frequency may not be the expected "
+            << "value and may lead to functional errors.\n"
+            << "    If you wish to override this failure, please set "
+            << "\"report_chk\" variable to \"false\" in <board_test.cpp> and "
+            << "recompile host code only using \"-reuse-exe=board_test.fpga\" "
+            << "option in compile command.\n"
+            << "    Please run complete board_test at least once and ensure "
+            << "the hardware frequency matches expected frequency, mismatch "
+            << "may lead to functional error.\n\n";
+
+  // **** Launching an empty kernel (no op) **** //
+
+  // ND Range of kernel to launch
+  constexpr size_t kTotalBytes =
+      128 * kMB;  // 128 MB - this is the min amount known to be available on
+                  // device memory (minimum on device - e.g. Cyclone V)
+  constexpr size_t kGlobalSize = kTotalBytes / (sizeof(unsigned));
+
+  auto e = q.submit([&](handler &h) {
+    // Global range (1 dimension)
+    constexpr size_t kN = kGlobalSize;
+    // Work group Size (1 dimension)
+    constexpr size_t kReqdWgSize = 32 * kKB;  // 32 KB
+    h.parallel_for<NopNDRange>(
+        nd_range<1>(range<1>(kN), range<1>(kReqdWgSize)), [=](auto id) 
+		[[cl::reqd_work_group_size(1, 1, kReqdWgSize)]]{});
```

Comment:
Does this need to be a parallel_for? Can we make it a single_task and get the same measurement?